tests/test_integration.py

```diff
@@ -326,3 +328,114 @@ def test_pex_path_in_pex_info_and_env():
     stdout, rc = run_simple_pex(pex_out_path, [test_file_path], env=env)
     assert rc == 0
     assert stdout == b'Success!\n'
+
+
+def test_interpreter_constraints_to_pex_info():
+  with temporary_dir() as output_dir:
+    # constraint without interpreter class
+    pex_out_path = os.path.join(output_dir, 'pex1.pex')
+    res = run_pex_command(['--disable-cache',
+      '--interpreter-constraints=">=2.7,<3"',
+      '-o', pex_out_path])
+    if sys.version_info[0] == 3:
+      assert res.return_code == 102
+    else:
+      res.assert_success()
+      pex_info = get_pex_info(pex_out_path)
+      assert '>=2.7,<3' == pex_info.interpreter_constraints
+
+    # constraint with interpreter class
+    pex_out_path = os.path.join(output_dir, 'pex2.pex')
+    res = run_pex_command(['--disable-cache',
+      '--interpreter-constraints="CPython>=2.7,<3"',
+      '-o', pex_out_path])
+    if sys.version_info[0] == 3 or hasattr(sys, 'pypy_version_info'):
+      assert res.return_code == 102
+    else:
+      res.assert_success()
+      pex_info = get_pex_info(pex_out_path)
+      assert 'CPython>=2.7,<3' == pex_info.interpreter_constraints
+
+
+@pytest.mark.skipif(NOT_CPYTHON_36)
+def test_interpreter_constraints_to_pex_info_py36():
+  with temporary_dir() as output_dir:
+    # constraint without interpreter class
+    pex_out_path = os.path.join(output_dir, 'pex1.pex')
+    res = run_pex_command(['--disable-cache',
+      '--interpreter-constraints=">=3"',
+      '-o', pex_out_path])
+    res.assert_success()
+    pex_info = get_pex_info(pex_out_path)
+    assert '>=3' == pex_info.interpreter_constraints
+
+    # constraint with interpreter class
+    pex_out_path = os.path.join(output_dir, 'pex2.pex')
+    res = run_pex_command(['--disable-cache',
+      '--interpreter-constraints="CPython>=3"',
+      '-o', pex_out_path])
+    res.assert_success()
+    pex_info = get_pex_info(pex_out_path)
+    assert 'CPython>=3' == pex_info.interpreter_constraints
+
+
+def test_resolve_interpreter_with_constraints_option():
+  with temporary_dir() as output_dir:
+    pex_out_path = os.path.join(output_dir, 'pex1.pex')
+    res = run_pex_command(['--disable-cache',
+      '--interpreter-constraints=">=2.7,<3"',
+      '-o', pex_out_path])
+    if sys.version_info[0] == 3:
+      assert res.return_code == 102
+    else:
+      res.assert_success()
+
+    pex_out_path = os.path.join(output_dir, 'pex2.pex')
+    res = run_pex_command(['--disable-cache',
+      '--interpreter-constraints=">3"',
+      '-o', pex_out_path])
+    if sys.version_info[0] == 3:
+      res.assert_success()
+    else:
+      assert res.return_code == 102
+
+
+@pytest.mark.skipif("hasattr(sys, 'pypy_version_info')")
+def test_interpreter_resolution_with_pex_python_path():
```

Comment:
This test case was tricky to figure out given the nature of the test and the need for multiple system interpreters that could vary between systems. I'm not sure if this will work in CI or if there is a better way to go about testing with dependence on system Python binaries, but I gave it my best shot and I am open to feedback/further iteration.

Reply:
I changed the interpreters for testing from system-specific Python installs to the Python binaries located in .tox directory. Hopefully this cleans up CI failures due to missing interpreters (when using the PythonInterpreter.all() approach).